### PR TITLE
Fix indent after multiline string

### PR DESCRIPTION
**Pull request by tompng**
This pullreq fixes two indent problem with multiline string.

### Fixed indent after `"stri\ng"`, `%[stri\ng]` `%w[string\nstring]`
```
# Case1
irb(main):001:0> if 0
irb(main):002:1"   "1
irb(main):003:1"   2"
irb(main):004:1* 3
expected:          3

# Case2
irb(main):001:1* if 0
irb(main):002:2"   p("1
irb(main):003:2"     2",
irb(main):004:2*   3,
expected:            3,
```

Multiline string can contain `"\n"`, but the next token does not matches `/(\A +)/`.
Skipped updating prev_spaces if token is :on_tstring_content or :on_words_sep

###  Fixed calculation of corresponding_token_depth after heredoc
```
irb(main):001:0> if 1
irb(main):002:1"   <<~A
irb(main):003:1"   A
irb(main):004:1*   p(
irb(main):005:3*     p(
irb(main):006:4*     )
irb(main):007:1* )
expected:          )
```
and after embdoc
```
irb(main):001:1* if 1
irb(main):002:1* =begin
irb(main):003:1* =end
irb(main):004:2*   p(
irb(main):005:3*     p(
irb(main):006:2*     )
irb(main):007:1* )
expected:          )
```

Added `:on_heredoc_end` and `:on_embdoc_end`. It also ends with `"\n"`.
```ruby
# all tokens that include "\n" in ruby/ruby source code
Dir.glob('**/*.rb').flat_map do |file|
  RubyLex.ripper_lex_without_warning(File.read(file), context: context).select { _1.tok.include? $/ }
end.map(&:event).uniq
# => [:on_comment, :on_nl, :on_ignored_nl, :on_tstring_content, :on_heredoc_end, :on_words_sep, :on_embdoc_beg, :on_embdoc, :on_embdoc_end, :on_sp, :on___end__]
```

### CODE REVIEW

File: test/irb/test_ruby_lex.rb [hunk 1]
```diff
@@ -632,5 +632,34 @@ def test_unterminated_code
...
+        lines = code.lines
+        lex.instance_variable_set('@tokens', RubyLex.ripper_lex_without_warning(code))
+        assert_equal 2, lex.check_corresponding_token_depth(lines, lines.size)
+      end
+    end
+
+    def test_find_prev_spaces_with_multiline_literal
+      lex = RubyLex.new
+      codes = [
+        "if true\n  1\n  hello\n  1\n  world\nend",
```
Comment (st0012) [inline test/irb/test_ruby_lex.rb:653]: I think to most people it's not easy to newline and spaces in their head. Do you think we can rewrite test cases into something like

```rb
            %[
if true
  1
  hello
  1
  world
end
            ].strip,
```

to improve its readability?

(Note: we can just put `strip` in the `each` block if you think the new format is better)

File: test/irb/test_ruby_lex.rb [hunk 2]
```diff
@@ -632,5 +632,34 @@ def test_unterminated_code
         assert_empty(error_tokens, 'Error tokens must be ignored if there is corresponding non-error token')
       end
     end
+
+    def test_corresponding_token_depth_with_heredoc_and_embdoc
+      codes = [
+        "if true\n  hello\n  p(\n    p(\n    )\n  )",
```
[the remark this answers is not in the document]
Comment (tompng) [reply test/irb/test_ruby_lex.rb]: I simplified it

File: test/irb/test_ruby_lex.rb (hunk 1 above)
Comment (tompng) [reply test/irb/test_ruby_lex.rb]: I agree that `"\n  \n  ..."` is not so readable. I rewrite it using `<<~EOC`.

**Review by st0012 — APPROVED** [review]
Sorry for the delay and thank you for the fix 👍

**Pull request merged**